docs/development.txt

```diff
@@ -0,0 +1,82 @@
+[[development-process]]
+JanusGraph Development Process
+==============================
+
+The following sections describe the JanusGraph development process.
+
+
+[[development-decisions]]
+Development Decisions
+---------------------
+
+Many development decisions will be made during the day-to-day work of JanusGraph contributors without any extra
+process overhead. However, for larger bodies of work and significant updates and additions, the following
+process shall be followed. Significant work may include, but is not limited to:
+
+* A major new feature or subproject eg. a new storage adapter
+* Incrementing the version of a core dependency such as a Apache TinkerPop
+* Addition or deprecation of a public API
+* Internal shared data-structure or API change
+
+For these sorts of changes, contributors will:
+
+* Create one or more issues in the Github issue tracker
+* Start a [DISCUSS] thread on the https://groups.google.com/forum/#!forum/janusgraph-dev[janusgraph-dev] list where the proposed change may be discussed by committers
+and other community members
+* When the proposer feels it appropriate, a VOTE shall be called. As with pull requests three +1 votes, including
+one from the relevant Technical Steering Committee (TSC) member if the change is limited to a single project, are required for the change to be accepted.
+
+Though not required, it is strongly recommended that contributors limit the amount of development work they
+perform before their change is accepted. In the instance where the change is not approved or significant updates to
+the proposal are made, we wish to minimize the amount of rework required.
+
+[[branching]]
+Branching
+---------
+For features that involve only one developer, developers will work in their own JanusGraph forks, managing their
+own branches, and submitting pull requests when ready. Feature branches in the main JanusGraph repository will be
+created when more than one developer collaborates on a feature.
+
+Branch Naming Conventions
+~~~~~~~~~~~~~~~~~~~~~~~~~
+All branch names shall be prepended with `Issue_#_`.
+
+[[pull-requests]]
+Pull Requests
+-------------
+
+Most changes to JanusGraph must be reviewed before they are committed (RTC). This workflow is performed using Github's
+pull request process. In instances where the committer deems the full process
+unnecessary, commit-then-review (CTR) may be employed. This should be used infrequently and only for extremely trivial
+changes. Changes of this sort may include:
+
+* Documentation typo or small documentation addition
+* Addition of a new test case
+
+Any commit that is made following CTR shall include the fact that it is a CTR in the commit comments.
```

Comment:
GitHub provides automated notifications: https://help.github.com/articles/managing-notifications-for-pushes-to-a-repository/

For example, we can create a list with the firehose of all commits / merges, e.g., janusgraph-commits@, and folks who want the firehose can subscribe to this. I've seen this done with the commit-then-review workflow in the LLVM project, it worked well.

I can set this up.